lib/ansible/config/helpers.py

```diff
@@ -0,0 +1,40 @@
+# Copyright: (c) 2017, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import annotations
+
+from ansible import constants as C
+from ansible.errors import AnsibleError
+from ansible.utils.display import Display
+
+display = Display()
+
+NOPES = frozenset(['$', '@', ':', '\\', '/', ';', '%', '{', '(', '}', ')', '"', "'", '`'])
+REQUIRED_UNIQUE = frozenset(['pid', 'timestamp'])
+REQUIRED_NAME = frozenset(['basename', 'stripname'])
+
+
+def get_validated_backup_file_name_template(tmplt: str) -> str:
+
+    try:
+        for required in REQUIRED_NAME:
+            if required in tmplt:
+                break
+        else:
+            raise AnsibleError("Required name variable (basename or stripname) not")
```

Comment:
Maybe avoid using a generic catch-all base exception and have something like `LookupError`? Especially, since it's being handled within the scope, anyway.